### PR TITLE
docs: Add docs for plugins structure

### PR DESCRIPTION
**Pull request by victorgcramos**
Docs for Plugin Architecture early development (https://github.com/decred/politeiagui/pull/2486)

TODO:

- [x] Ticketvote Docs
- [x] App Setup docs
- [x] App Shell
- [x] Plugin integration

### CODE REVIEW

File: plugins-structure/README.md [hunk 1]
```diff
@@ -0,0 +1,579 @@
+# PoliteiaGUI - Plugins Structure
+
+A Guest User Interface configurable structure for creating multiple applications
+using Politeia's [[1]](#1) plugin architecture.
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]: You are referencing a book that's behind a paywall. That's not good because the reader doesn't have free access to the resource.

File: plugins-structure/README.md [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+
+The following guide will assume you are familiar with the tools listed below. We
+recommend you to take some time to study them before proceeding.
+
+- [JavaScript](https://developer.mozilla.org/docs/Web/JavaScript)
+- [Micro-Frontends](https://micro-frontends.org/)
+- [Lerna](https://lerna.js.org/)
+- [Yarn Workspaces](https://yarnpkg.com/features/workspaces)
+- [Politeia](https://github.com/decred/politeia)
+- [Redux Toolkit](https://redux-toolkit.js.org/)
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md:24]: I would add redux to this list.

File: plugins-structure/README.md [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+- [Window History](https://developer.mozilla.org/docs/Web/API/Window/history)
+
+## Introduction
+
+Politeia is the underlying platform that provides a versioned, timestamped data
+store [[2]](#2), which can be used for creating new applications on top of it.
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]: I don't think you have to link the politeia dev proposal here since you are doing it right below already.

File: plugins-structure/README.md [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+However, the stored data could be any set of data, not exclusively proposals,
+which could also be used as a more generic storage, such as contractors
+payments management, proposals discussions, etc.
+
+The current design would allow developers to create new plugins to interact,
+extend and manage the Record functionality. Therefore, we can use Politeia
+[[1]](#1) architecture to create more applications, such as
+[Decred's Proposal system (Pi)](https://proposals.decred.org) and
+[Decred's Contractor Management System (CMS)](https://cms.decred.org/)
+_(CMS not implemented on the new architecture yet)_.
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]: I think these paragraphs doesn't add much to the text itself. I would rephrase them to something like:

`A lot of applications can benefit from this design, that's why we decided to create a composable plugin architecture. Developers are now able to create plugins to interact, extend and manage the records stored and create awesome applications with it.`

File: plugins-structure/README.md [hunk 5]
```diff
@@ -0,0 +1,579 @@
...
+provide a reusable structure for multiple applications, which will provide a
+better code maintainance and scalability.
+
+On the Frontend side, it means we can provide different UXs depending on the
+plugins combinations. We can integrate multiple plugins for customizing some
+app, and still not trigger collateral effects on other plugins.
+
+## GUI Composable Plugin Architecture
+
+The GUI architecture is a mono-repository structure, composed by
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md:66]: I don't think we could define our structure as micro-frontends. We use a lot of the same concepts but micro-frontends have independent deploy pipelines and micro-apps are combined on run-time. I'd rather we don't call our `plugins` micro-frontends and just stick with plugins.

File: plugins-structure/README.md [hunk 6]
```diff
@@ -0,0 +1,579 @@
...
+<p align="center">
+  <img src="../docs/assets/ps-overview-politeia.svg" />
+</p>
+
+Politeia is the application-shell. Pi, Ticketvote, Comments are plugins that
+extend the core Record package.
+
+> The User layer is currently under development and you can follow its progress
+> on [(decred/politeia#1479) politeiaww: Add user
+> plugins](https://github.com/decred/politeia/issues/1479).
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]: I don't think we have to mention the under development stage of the user layer because the pgui plugin architecture will be merge only after its completion.

File: plugins-structure/README.md [hunk 7]
```diff
@@ -0,0 +1,579 @@
...
+  adding new features, new routes, and also a custom UI.
+
+The following diagram displays an overview of the regarded plugin architecture
+for the proposals (Politeia) application:
+
+<p align="center">
+  <img src="../docs/assets/ps-overview-politeia.svg" />
+</p>
+
+Politeia is the application-shell. Pi, Ticketvote, Comments are plugins that
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]: I don't think we would have a plugin called `Pi`, shouldn't it be `Records` instead? All filenames and configurations would be passed down by the `Politeia` application-shell.

File: plugins-structure/README.md [hunk 8]
```diff
@@ -0,0 +1,579 @@
...
+  integration, in our case, plugins and core, into one user application.
+- Packages: workspace for plugins. Each **plugin** has its own isolated config,
+  and can compose the application shell by extending the **core** package and
+  adding new features, new routes, and also a custom UI.
+
+The following diagram displays an overview of the regarded plugin architecture
+for the proposals (Politeia) application:
+
+<p align="center">
+  <img src="../docs/assets/ps-overview-politeia.svg" />
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]: I think here you should include the `Records` plugin.

File: plugins-structure/README.md [hunk 9]
```diff
@@ -0,0 +1,579 @@
...
+The application shell is the bridge between the application user and the
+plugins. It implements UX-oriented features, and specific setup for some regular
+app. In the example displayed above, Politeia is responsible for integrating the
+multiple plugins, by adding wrappers around the plugins and core tools, and
+providing a custom user interface.
+
+The following diagram illustrates the Proposal Details page on the Politeia app:
+
+<p align="center">
+  <img src="../docs/assets/ps-app-shell-politeia.svg" />
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]: 👍 (this screenshot could be improved tho)

File: plugins-structure/README.md [hunk 10]
```diff
@@ -0,0 +1,579 @@
...
+</p>
+
+Politeia is the proposals system, and, in this case, proposals are Records that
+include proposal metadata files. Each proposal also has its own comments tree,
+therefore, it can combine Records and Comments. Also, users can check voting
+details for proposals due to the inclusion of the Ticketvote plugin, update the
+billing status and submit author updates for approved and active proposals
+adding the Pi plugin.
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]: This paragraph is a little confusing.

File: plugins-structure/README.md [hunk 11]
```diff
@@ -0,0 +1,579 @@
...
+
+Notice that the Icon, the dark-mode toggle, "Go Back" link, the page layout
+(components order, screen display) are controlled by the application shell,
+and, as mentioned before, are configured focusing on the UX.
+
+We can also create new applications with different combinations of plugins. One
+example could be CMS, which could use invoices as the application records, also
+implement comments for each invoice, use custom contractor management tools with
+an hypothetical CMS plugin, and other stuff.
```
Comment (tiagoalvesdulce) [inline plugins-structure/README.md]:
```suggestion
We could also create applications with different combinations of plugins. One good example is the Contractor Management System (CMS). CMS could use invoices as application records, also implement comments for each invoice and create new plugins to add custom functionality.
```

**Review by tiagoalvesdulce — COMMENTED** [review]
Nice work! Left a few inline comments. Also, I think you can move your assets to a docs folder inside the `plugins-architecture` folder.

File: plugins-structure/README.md (hunk 9 above)
Comment (victorgcramos) [reply plugins-structure/README.md]: Nice, will do!

**Review by tiagoalvesdulce — APPROVED** [review]
LGTM! Great job
